### PR TITLE
Fixes disposal bin, DirectionalPassable, Disposal intake, pushing in editor

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Fixes being unable to drag things into the disposal bin.
- Fixes `DirectionalPassable` component.
    - Fixes disposal intakes unable to be dragged from any direction other than its opening. Fixes part 21. of issue #4654.
- Swaps directional windows to use `DirectionalPassable` instead of `RegisterDoor`.
- Fixes pushing while playing in editor being buggy.

> Note: it is now possible to convert directional doors to use `DirectionalPassable`, too.

CL:Fixes being unable to dispose of yourself, and unable to drag disposal intakes.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs [hunk 1]
```diff
@@ -1,105 +1,210 @@
...
+					break;
+				case PassType.Leaving:
+					isLeavableOnAll = false;
+					Passable = false;
+					break;
+				case PassType.Atmospheric:
+					isAtmosPassableOnAll = false;
+					AtmosPassable = false;
+					break;
+				default:
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs:124]: Codacy found an issue: [Remove this empty 'default' clause.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637)

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs  1
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Objects/PushPull.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637&bid=20689751)

**Comment by corp-0** [comment]
If you do update doors to use this component, remember to update the new doors instead

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs  1
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Objects/PushPull.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637&bid=20689751)

File: UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs [hunk 2]
```diff
@@ -1,105 +1,204 @@
...
+		}
+
+		#endregion RegisterObject Overrides
+
+		/// <summary>
+		/// Sets all sides for this object as passable for the given pass type.
+		/// </summary>
+		public void AllowPassableOnAllSides(PassType passType)
+		{
+			switch (passType)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs:86]: Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637)

**Comment by NoooneyDude** [comment]
`Codacy`: remove the empty default clause.
`Codacy`: add a default clause.

But what if I just want to do nothing, `Codacy`?

**Comment by corp-0** [comment]
change the name of the pr 👀

**Comment by corp-0** [comment]
> Codacy: remove the empty default clause.
Codacy: add a default clause.

if you ever get to the default, then something went wrong with the values because you don't really wanted to get to that clause. You can log.error it so we know for some reason the switch didn't get the values we were expecting. For that reason having a default clause is a sensible decision.

**Comment by Bod9001** [comment]
just need to add the log statement will merge after that

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Objects/Directionals/DirectionalPassable.cs  1
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Objects/PushPull.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6377637&bid=20689751)

**Comment by Bod9001** [comment]
is all good to merge now?

**Comment by NoooneyDude** [comment]
Yes good to merge!

**Pull request merged**